numpy/f2py/src/fortranobject.c

```diff
@@ -590,6 +590,11 @@ static void f2py_report_on_array_copy_fromany(void) {
  * $Id: fortranobject.c,v 1.52 2005/07/11 07:44:20 pearu Exp $
  */
 
+static int check_and_fix_dimensions(const PyArrayObject* arr,
+                                    const int rank,
+                                    npy_intp *dims,
+                                    char *msgbuffer, int msgbuffer_len);
```

Comment:
Yes, that would work too.  I probably won't get back to this for several days (where "several" is drawn from a random distribution with a fat tail).